### PR TITLE
experimental/diff: move isMainModule block to tests/stdlib/tdiff.nim

### PR DESCRIPTION
**Pull request by timotheecour**
view diff with: `git show --color-moved --color-moved-ws=ignore-all-space HEAD`

CI failure unrelated: https://github.com/nim-lang/Nim/issues/16169

### CODE REVIEW

**Comment by ringabout** [comment]
Yeah, I will clean `assert => doAssert` soon.

File: tests/stdlib/tdiff.nim [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import experimental/diff
+import std/strutils
+
+proc testHelper(f: seq[Item]): string =
+  for it in f:
+    result.add(
+      $it.deletedA & "." & $it.insertedB & "." & $it.startA & "." & $it.startB & "*"
+    )
+
+template main() =
```
Comment (ringabout) [inline tests/stdlib/tdiff.nim]: Why change to `templates`?

**Comment by ringabout** [comment]
The rest looks good to me.
image
https://www.diffchecker.com/pfhO4juY

File: tests/stdlib/tdiff.nim (hunk 1 above)
Comment (timotheecour) [reply tests/stdlib/tdiff.nim]: changed it back

**Review by ringabout — APPROVED** [review]

**Comment by juancarlospaco** [comment]
This module should be moved out of that experimental folder that only exists for it,
 is not too unstable and code looks good enough.

**Comment by timotheecour** [comment]
> This module should be moved out of that experimental folder that only exists for it,
is not too unstable and code looks good enough.

that would require improving API first eg

```nim
newDiffData => initDiffData
sms => shortestMiddleSnake

# and code cleanups eg:

# Debug.Write(0, "sms", "extend reverse path " + k.ToString())
=>
when defined(nimDiffDebug): ...
```
* allowing a generic API

* using runnableExamples
* and many other things

it's doable after that

**Pull request merged**